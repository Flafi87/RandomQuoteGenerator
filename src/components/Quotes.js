import React from 'react';
import { Card, Button } from 'reactstrap';
import PropTypes from 'prop-types';

const Quotes = ({quote, author, handleClick}) => {
return(
  <Card id="quote-box">
    <div id="text">
      <h3>
      &quot;
        {quote}
      &quot;
      </h3>

    </div>
    <div id="author">
      <h4>
-
        {author}
      </h4>
    </div>
    <div className="buttons">
      <a
        href={`https://twitter.com/intent/tweet?hashtags=quotes&related=freecodecamp&text="${quote}" ${author}`}
        id="tweet-quote"
        title="Tweet this quote!"
      >
        <i className="fa fa-twitter" />
      </a>
      <Button
        type="button"
        id="new-quote"
        onClick={handleClick
        }
      >
        NEW QUOTE
      </Button>
    </div>
  </Card>
)
}

Quotes.propTypes = {
    quote: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired
}

export default Quotes