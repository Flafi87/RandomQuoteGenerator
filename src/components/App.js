import React from "react";
import axios from "axios";
import { Button, Spinner } from "reactstrap";
import Quotes from "./Quotes";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      quotes: {},
      myQuote: {},
    };
  }

  componentDidMount() {
    this.fetchaQuote();
  }

  fetchaQuote = () => {
    axios
      .get(
        "https://gist.githubusercontent.com/camperbot/5a022b72e96c4c9585c32bf6a75f62d9/raw/e3c6895ce42069f0ee7e991229064f167fe8ccdc/quotes.json"
      )
      .then((res) => {
        this.setState({ quotes: res.data.quotes });
        this.setState({
          myQuote: res.data.quotes[Math.floor(Math.random() * 102)],
        });
      });
  };

  handleClick = () => {
    const { quotes } = this.state;
    this.setState({
      myQuote: quotes[Math.floor(Math.random() * 102)],
    });
  };

  render() {
    const { myQuote } = this.state;
    const { quote, author } = myQuote;
    const spinner = (
      <div className="d-flex justify-content-center align-middle">
        <Spinner color="primary" />
      </div>
    );
    return (
      <div className="container" bg="dark">
        <div className="d-flex justify-content-between">
          <Button
            className="my-3"
            color="success"
            href="https://flafi.me/index.html#jscript"
          >
            Back to the website
          </Button>
          <Button
            className="my-3"
            color="primary"
            href="https://github.com/Flafi87/RandomQuoteGenerator"
            target="_blank"
          >
            Repo
          </Button>
        </div>

        {quote ? (
          <>
            <div className="d-flex justify-content-center align-middle">
              <Button type="button" id="new-quote" onClick={this.handleClick}>
                NEW QUOTE
              </Button>
            </div>
            <Quotes
              handleClick={this.handleClick}
              quote={quote}
              author={author}
            />
          </>
        ) : (
          spinner
        )}
      </div>
    );
  }
}

export default App;
