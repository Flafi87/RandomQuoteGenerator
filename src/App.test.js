import React from "react";
import { create } from "react-test-renderer";
import Quotes from "./components/Quotes";
import App from "./components/App";

// eslint-disable-next-line prettier/prettier
describe('Quotes snapshot test', () => {
  test("testing Quotes", () => {
    const tree = create(<Quotes />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("App snapshot test", () => {
  test("testing App", () => {
    const tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
