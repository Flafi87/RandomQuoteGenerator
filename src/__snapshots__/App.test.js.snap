// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test testing App 1`] = `
<div
  className="container"
>
  <div
    className="d-flex justify-content-between"
  >
    <a
      aria-label={null}
      className="my-3 btn btn-success"
      href="https://flafi.me/index.html#jscript"
      onClick={[Function]}
    >
      Back to the website
    </a>
    <a
      aria-label={null}
      className="my-3 btn btn-primary"
      href="https://github.com/Flafi87/RandomQuoteGenerator"
      onClick={[Function]}
      target="_blank"
    >
      Repo
    </a>
  </div>
  <div
    className="d-flex justify-content-center align-middle"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`My first snapshot test testing Quotes 1`] = `
<div
  className="card"
  id="quote-box"
>
  <div
    id="text"
  >
    <h3>
      "
      "
    </h3>
  </div>
  <div
    id="author"
  >
    <h4 />
  </div>
  <div
    className="buttons"
  >
    <a
      href="https://twitter.com/intent/tweet?hashtags=quotes&related=freecodecamp&text=\\"undefined\\" undefined"
      id="tweet-quote"
      title="Tweet this quote!"
    >
      <i
        className="fa fa-twitter"
      />
    </a>
    <button
      aria-label={null}
      className="btn btn-secondary"
      id="new-quote"
      onClick={[Function]}
      type="button"
    >
      NEW QUOTE
    </button>
  </div>
</div>
`;

exports[`My second snapshot test testing App 1`] = `
<div
  className="container"
>
  <div
    className="d-flex justify-content-between"
  >
    <a
      aria-label={null}
      className="my-3 btn btn-success"
      href="https://flafi.me/index.html#jscript"
      onClick={[Function]}
    >
      Back to the website
    </a>
    <a
      aria-label={null}
      className="my-3 btn btn-primary"
      href="https://github.com/Flafi87/RandomQuoteGenerator"
      onClick={[Function]}
      target="_blank"
    >
      Repo
    </a>
  </div>
  <div
    className="d-flex justify-content-center align-middle"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`Quotes snapshot test testing Quotes 1`] = `
<div
  className="card"
  id="quote-box"
>
  <div
    id="text"
  >
    <h3>
      "
      "
    </h3>
  </div>
  <div
    id="author"
  >
    <h4 />
  </div>
  <div
    className="buttons"
  >
    <a
      href="https://twitter.com/intent/tweet?hashtags=quotes&related=freecodecamp&text=\\"undefined\\" undefined"
      id="tweet-quote"
      title="Tweet this quote!"
    >
      <i
        className="fa fa-twitter"
      />
    </a>
    <button
      aria-label={null}
      className="btn btn-secondary"
      id="new-quote"
      onClick={[Function]}
      type="button"
    >
      NEW QUOTE
    </button>
  </div>
</div>
`;
